src/app/components/AnalyticsConsent/initializeMatomo.ts

```diff
@@ -0,0 +1,88 @@
+window._paq = window._paq || []
+export const matomoDomain = 'https://matomo.oasis.io/'
+window._paq.push(['setSiteId', '3'])
+// https://developer.matomo.org/guides/tracking-consent#step-1-require-consent
+// https://matomo.org/faq/general/faq_156/
+window._paq.push(['requireConsent'])
+window._paq.push(['setDoNotTrack', true])
+window._paq.push(['enableLinkTracking'])
+window._paq.push(['setTrackerUrl', `${matomoDomain}matomo.php`])
+
+// https://developer.matomo.org/guides/tracking-javascript-guide#tracking-one-domain-and-its-subdomains-in-the-same-website
+// https://matomo.org/faq/how-to/faq_23654/
+// window._paq.push(['setDomains', ['*.domain1.com', '*.domain2.com']])
+// window._paq.push(['enableCrossDomainLinking'])
+// window._paq.push(['setCookieDomain', '*.example.org'])
+
+window._paq.push([
+  'setCustomRequestProcessing',
+  (queryString: string) => {
+    const params = new URLSearchParams(queryString)
+    for (const name of ['action_name', 'url', 'urlref']) {
+      const v = params.get(name)
+      if (v) {
+        params.set(name, redactURL(v))
+      }
+    }
+
+    return params.toString()
+  },
+])
+
+export function loadMatomo() {
+  const d = document,
+    g = d.createElement('script'),
+    s = d.getElementsByTagName('script')[0]
+  g.async = true
+  // TODO: Enable SRI and fix CORS on matomo.js to support SRI.
+  // Warning: https://github.com/matomo-org/matomo/commits/5.x-dev/js/piwik.min.js
+  // changed 5 times between 2023 Feb and 2024 Jan.
+  // g.integrity = 'sha384-yGkhmoBpWPCg9IZSGIz64itCivgzaXu+CXflD1GthTMPW3YUjF4FoJ8YUBGDTOF7'
+  g.src = `${matomoDomain}matomo.js`
+  s.parentNode?.insertBefore(g, s)
+}
+
+export function optInForOneYear() {
+  // https://developer.matomo.org/guides/tracking-consent#a-if-you-want-to-let-matomo-remember-the-consent
+  window._paq.push(['rememberConsentGiven', 365 * 24])
+}
+
+export function decline() {
+  window._paq.push(['forgetConsentGiven'])
+}
+
+export type HasAccepted = 'opted-in' | 'declined' | 'never-responded' | 'error_matomo_not_loaded'
+export function hasAccepted(): HasAccepted {
+  let response: HasAccepted = 'error_matomo_not_loaded'
+  window._paq.push([
+    function () {
+      if (this.hasRememberedConsent()) {
+        response = 'opted-in'
+      } else {
+        response = 'declined'
+
+        // Can not differentiate between declined and never-responded without workaround
+        // Based on https://github.com/matomo-org/matomo/blob/082782b/js/piwik.js#L7176-L7192
+        // TODO: remove workaround after https://github.com/matomo-org/matomo/issues/17169 is solved
+        const CONSENT_COOKIE_NAME = 'mtm_consent'
+        const CONSENT_REMOVED_COOKIE_NAME = 'mtm_consent_removed'
+        if (!getCookieWorkaround(CONSENT_COOKIE_NAME) && !getCookieWorkaround(CONSENT_REMOVED_COOKIE_NAME)) {
+          response = 'never-responded'
+        }
+      }
+    },
+  ])
+  return response
+}
+
+function getCookieWorkaround(cookieName: string) {
```

Comment:
Does cookie name changes? Do we have any control over that on BE side?

Reply:
Cookie seems to only be used on the frontend https://matomo.oasis.io/matomo.js

Reply:
Disregard my original comment, I was confused because I am used to different `getCookie` snippet. Can we just call the function `getCookie`? Not sure why `Workaround` is mentioned?

Reply:
I named it "*Workaround" because I'd like to delete this regex implementation if we don't need it for matomo anymore. I don't think this regex implementation is very safe and robust and probably shouldn't be reused by other code.
At the same time, I'm not sure if using a different more safe implementation could introduce discrepancies with matomo internals.

Reply:
Got it. If you think it is unsafe, I would name the function `getCookieUnsafe`, workaround naming seems weird to me.

Either way, I like this piece of code with Regex. You need to parse cookies somehow. And I prefer this piece to be included in the code instead of using external library - that would add unnecessary footprint. I would just move it to `CookiesUtils` or something similar.

Reply:
Made an upstream issue about the regex implementation: https://github.com/matomo-org/matomo/issues/21998